easybuild/easyconfigs/w/waLBerla/waLBerla-6.1-foss-2023a.eb

```
@@ -0,0 +1,48 @@
+# easyconfig file for waLBerla v6.1
+easyblock = 'CMakeMake'
+
+name = 'waLBerla'
+version = '6.1'
+
+homepage = "https://walberla.net/index.html"
+description = """Widely applicable Lattics-Boltzmann from Erlangen is a 
+block-structured high-performance framework for multiphysics simulations"""
+
+toolchain = {'name': 'foss', 'version': '2023a'}
+toolchainopts = {'usempi': True, 'pic': True}
+
+source_urls = ['https://i10git.cs.fau.de/walberla/walberla/-/archive/v6.1']
+sources = ['%(name)s-v%(version)s.tar.gz']
+patches = ['waLBerla-6.1_fix_cmakelist_for_easybuild.patch']
+checksums = [
+    {'waLBerla-v6.1.tar.gz': 'f0acdd9ad6543bc9306c8aae953dd5065986271d4398916ae0469db8b21c007a'},
+    {'waLBerla-6.1_fix_cmakelist_for_easybuild.patch':
+     '057066d07294e8f159c2545d2d5f9138d439dbb54bc70eb75eca1f82ae650c47'},
+]
+
+builddependencies = [('CMake', '3.26.3')]
+
+dependencies = [
+    ('pybind11', '2.11.1'),
+    ('Python', '3.11.3'),
+    ('Boost.MPI', '1.79.0'),
+    ('SciPy-bundle', '2023.07'),
+]
+
+parallel = 1
+
+configopts = "-DWALBERLA_BUILD_WITH_PYTHON=ON "
+configopts += "-DWALBERLA_BUILD_SHOWCASES=OFF "
+configopts += "-DWALBERLA_BUILD_DOC=OFF "
+configopts += "-DPython_ROOT_DIR=$EBROOTPYTHON "
+
+modextrapaths = {'PYTHONPATH': ['pythonmodule']}
+
+sanity_check_paths = {
+    'files': ['waLBerlaDefinitions.h', 'pythonmodule/walberla_cpp.cpython-311-x86_64-linux-gnu.so'],
```

I had overlooked this in the other waLBerla PR (https://github.com/easybuilders/easybuild-easyconfigs/pull/18932), and we still need to test it on Arm systems in the EESSI context anyway, but this will need to be changed in order to work on Arm. We could probably use the `%(arch)s` template value for this, assuming only the `x86_64` part in the filename will change:
```suggestion
    'files': ['waLBerlaDefinitions.h', 'pythonmodule/walberla_cpp.cpython-311-%(arch)s-linux-gnu.so'],
```

---

Sounds good, I'm testing it now and will push if no more changes are needed.